go/src/koding/klient/app/klient.go

```diff
@@ -461,11 +465,11 @@ func (k *Klient) RegisterMethods() {
 	k.kite.HandleFunc("klient.usage", k.usage.Current)
 
 	// klient os method(s)
-	k.kite.HandleFunc("os.home", kos.Home)
-	k.kite.HandleFunc("os.currentUsername", kos.CurrentUsername)
+	k.handleWithSub("os.home", kos.Home)
```

Comment:
Honestly speaking, the current form was based on your solution for `remote` methods and I'm not against other ideas. Personally, I would have written something like:

```go
package subscription

type Check struct {
    Team func() (team.Team error) // team getter.
}

func (c *Check) Check() {
  // business logic here
}

// ...

func KiteCheck(c *Check, h kite.Handler) kite.Handler {
  return kite.HandlerFunc(func(req *Request) (result interface{}, err error) {
    if err := c.Check(); err != nil { // validation logic
      return nil, err
    }
    return h(req)
  })
}

```
and 
```go
k.kite.Handle("os.currentUsername", subscription.KiteCheck(c, kite.HandlerFunc(kos.CurrentUsername)))
```
With this, I could test the logic separately from klient/kite. But, I know that I sometimes overdo with my architecture ideas so I'm trying to not push them on anyone. :)